File: dangling_pointer.c
#include <stdio.h>

int *dangling_address()
{
    int a = 10;
    return &a;
}

int main()
{
    int *ptr;
    ptr = dangling_address();

    printf(
        "ptr = %p (%d)\n",
        (void *) ptr, *ptr
    );
}

File: dangling_pointer.rs


fn dangling_address() -> &'static i32
{
    let a = 10;
    &a
}

fn main()
{
    let ptr: &i32;
    ptr = dangling_address();

    println!(
        "ptr = {:p} ({})",
        ptr, *ptr
    );
}
